examples/DrawXNative/DrawXShared/RealmDraw.cs

```diff
@@ -0,0 +1,572 @@
+                _hasSavedBitmap = true;
+            } // SKPaint
+
+            _redrawPathsAtNextDraw = false;
+        }
+
+        public void StartDrawing(float inX, float inY)
+        {
+            _currentlyDrawing = null;  // don't clear in Stop as will lose last point, clear when we know done
+            if (TouchInControlArea(inX, inY))
+            {
+                _ignoringTouches = true;
+                return;
+            }
+
+            _ignoringTouches = false;
+            if (_realm == null)
+            {
+                if (!_waitingForLogin)
+                {
+                    LoginToServerAsync();
+                }
+
+                return;  // not yet logged into server, let next touch invoke us
+            }
+
+            _lastX = inX;
+            _lastY = inY;
+            Debug.WriteLine($"Writing a new path starting at {inX}, {inY}");
+
+            // start a local path for responsive drawing
+            _currentlyDrawing = new SKPath();
+            _currentlyDrawing.MoveTo(inX, inY);
+
+            ScalePointsToStore(ref inX, ref inY);
+            _isDrawing = true;
+            _realm.Write(() =>
+            {
+                _drawPath = new DrawPath() { color = CurrentColor.Name };  // Realm saves name of color
+                _drawPath.points.Add(new DrawPoint() { x = inX, y = inY });
+                _realm.Add(_drawPath);
+            });
+            /*            var ignoredTask = _realm.WriteAsync((asyncRealm) =>
+                        {
+                            _drawPath = new DrawPath() { color = currentColor.Name, NumPointsDrawnLocally = 1 };  // Realm saves name of color
+                            _drawPath.points.Add(new DrawPoint() { x = inX, y = inY });
+                            asyncRealm.Add(_drawPath);
+                        });*/
+        }
+
+        public void AddPoint(float inX, float inY)
+        {
+            if (_ignoringTouches)
+            {
+                return;  // probably touched in pencil area
+            }
+
+            if (_realm == null)
+            {
+                return;  // not yet logged into server
+            }
+
+            if (!_isDrawing)
+            {
+                // has finished connecting to Realm so this is actually a start
+                StartDrawing(inX, inY);
+                return;
+            }
+
+            _lastX = inX;
+            _lastY = inY;
+            Debug.WriteLine($"Adding a point at {inX}, {inY}");
+            _currentlyDrawing.LineTo(inX, inY);
+
+            ScalePointsToStore(ref inX, ref inY);
+            _realm.Write(() =>
+            {
+                _drawPath.points.Add(new DrawPoint() { x = inX, y = inY });
+            });
+            /*            _realm.WriteAsync((asyncRealm) =>
+                        {
+                            _drawPath.points.Add(new DrawPoint() { x = inX, y = inY });
```

Comment:
It seems you're not using the `asyncRealm`, so that is likely the reason it was not working.